src/main/kotlin/no/nav/syfo/personstatus/infrastructure/database/repository/PersonOversiktStatusRepository.kt

```diff
@@ -328,7 +329,14 @@ class PersonOversiktStatusRepository(private val database: DatabaseInterface) :
             is Search.ByDate -> searchByDate(search)
             is Search.ByInitialsAndDate -> searchByInitialsAndDate(search)
         }
-        return results.map { it.toPersonOversiktStatus() }
+        val personOppfolgingstilfelleVirksomhetMap = database.connection.use { connection ->
+            connection.getPersonOppfolgingstilfelleVirksomhetMap(pPersonOversikStatusIds = results.map { it.id })
```

Comment:
Gjenbruker bulk-spørring vi har for å hente `PPersonOppfolgingstilfelleVirksomhet` for en liste med personer (`SELECT *
    FROM PERSON_OPPFOLGINGSTILFELLE_VIRKSOMHET
    WHERE person_oversikt_status_id = ANY(?)`)

Et alternativ kunne kanskje vært å joine med `PERSON_OPPFOLGINGSTILFELLE_VIRKSOMHET` på et vis i selve søke-sqlen, men litt usikker på om det blir noe enklere og om det vil være bedre ytelsesmessig.

En annen mulig ytelsesforbedring kunne vært å gjøre dette _etter_ vi har filtrert bort de veileder ikke har tilgang til, men det er litt mer knot å få til kodemessig (siden vi gjør tilgangssjekk i laget utenfor her etter vi har laget domeneobjektene). Tror heller ikke antall resultater man får på søket i databasen er så stort at det vil bli et problem.

Reply:
Average execution time for det query'et er 3.58 ms, så det burde gå bra.